File: app/src/main/java/co/tpcreative/supersafe/model/EventItem.java
package co.tpcreative.supersafe.model;
import androidx.annotation.NonNull;

public class EventItem extends ListItem {

	@NonNull
	private Event event;
	@NonNull EnumEvent enumEvent;


	public EventItem(@NonNull Event event,@NonNull EnumEvent enumEvent) {
		this.event = event;
		this.enumEvent = enumEvent;
	}

	@NonNull
	public Event getEvent() {
		return event;
	}

	// here getters and setters
	// for title and so on, built
	// using event

	@Override
	public int getType() {
		return TYPE_EVENT;
	}

	@Override
	public EnumEvent getTypeEvent() {
		return enumEvent;
	}

}

File: app/src/main/java/co/tpcreative/supersafe/model/EventItem.kt
package co.tpcreative.supersafe.model

class EventItem(private val event: Event, enumEvent: EnumEvent) : ListItem() {
    var enumEvent: EnumEvent
    fun getEvent(): Event {
        return event
    }

    // here getters and setters
    // for title and so on, built
    // using event
    override fun getType(): Int {
        return ListItem.Companion.TYPE_EVENT
    }

    override fun getTypeEvent(): EnumEvent? {
        return enumEvent
    }

    init {
        this.enumEvent = enumEvent
    }
}